pkg/storage/tpr/storage_interface.go

```diff
@@ -325,31 +325,52 @@ func (t *store) List(
 		return err
 	}
 
-	req := t.cl.Get().AbsPath(
-		"apis",
-		groupName,
-		tprVersion,
-		"namespaces",
-		ns,
-		t.singularKind.URLName(),
-	)
-
-	var unknown runtime.Unknown
-	if err := req.Do().Into(&unknown); err != nil {
-		glog.Errorf("doing request (%s)", err)
-		return err
+	if t.hasNamespace && ns == t.defaultNamespace {
+		// if the resource is supposed to have a namespace, and the given one is the default,
```

Comment:
got it. no, this isn't the behavior of built-in objects, because they don't have the concept of a "default namespace" (because they are stored in etcd).